tests/cnf/core/network/metallb/tests/bgp-gracefulrestart.go

```diff
@@ -127,6 +144,157 @@ var _ = Describe("BGP Graceful Restart", Ordered, Label(tsparams.LabelGRTestCase
+			})
+			Expect(err).ToNot(HaveOccurred(), "Failed to list pods")
+			bfdProfile = createBFDProfileAndVerifyIfItsReady(frrk8sPods)
+
+			By("Creating MetalLb configMap")
+			bfdConfigMap = createConfigMap(tsparams.LocalBGPASN, ipv4NodeAddrList, false, true)
+		})
+		Context("single hop", func() {
+			BeforeEach(func() {
+				By("Creating BGP peer")
+				bgpPeer, err = metallb.NewBPGPeerBuilder(APIClient, "testpeer", NetConfig.MlbOperatorNamespace,
+					ipv4MetalLbIPList[0], tsparams.LocalBGPASN, tsparams.LocalBGPASN).
+					WithBFDProfile(bfdProfile.Definition.Name).WithPassword(tsparams.BGPPassword).
+					WithGracefulRestart(true).Create()
+				Expect(err).ToNot(HaveOccurred(), "Fail to create bgpPeer")
+
+				By("Creating FRR Pod with network and IP address")
+				frrPod = createFrrPod(
+					masterNodeList[0].Object.Name, bfdConfigMap.Object.Name, []string{}, staticIPAnnotation)
+
+				By("Checking that BGP and BFD sessions are established and up")
+				verifyMetalLbBFDAndBGPSessionsAreUPOnFrrPod(frrPod, ipv4NodeAddrList)
+
+				By("Checking that BGP GracefulRestart is enabled on the neighbors")
+				verifyGREnabledOnNeighbors(frrPod, ipv4NodeAddrList)
+			})
+
+			AfterEach(func() {
+				By("Applying the MachineConfig to remove the created nft policy")
+				createMCAndWaitforMCPStable(tsparams.DeleteAllRules)
+				err := mco.NewMCBuilder(APIClient, "98-nft").Delete()
+				Expect(err).ToNot(HaveOccurred(), "Failed to get the machineConfig")
+
+				By("Checking that BGP and BFD sessions are established and up")
+				verifyMetalLbBFDAndBGPSessionsAreUPOnFrrPod(frrPod, ipv4NodeAddrList)
+			})
+
+			It("Test Compatibility with BFD", reportxml.ID("77101"), func() {
+				gracefulRestartTestWithBFD("Cluster", addressPool[0], ipv4MetalLbIPList[0], frrPod, ipAddressPool, workerLabelMap)
+			})
+		})
+		Context("multi hop", func() {
+			var (
+				speakerRoutesMap  map[string]string
+				masterClientPodIP = "172.16.0.1"
+				frrMasterIPs      = []string{"172.16.0.253", "172.16.0.254"}
+			)
+			BeforeEach(func() {
+				By("Collecting information before test")
+				frrk8sPods, err := pod.List(APIClient, NetConfig.MlbOperatorNamespace, metav1.ListOptions{
+					LabelSelector: tsparams.FRRK8sDefaultLabel,
+				})
+				Expect(err).ToNot(HaveOccurred(), "Failed to list frrk8s pods")
+
+				speakerRoutesMap, err = buildRoutesMap(frrk8sPods, ipv4MetalLbIPList)
+				Expect(err).ToNot(HaveOccurred(), "Failed to build frrk8s route map")
+
+				By("Creating internal NAD")
+				masterBridgePlugin, err := nad.NewMasterBridgePlugin("internalnad", "br0").
+					WithIPAM(nad.IPAMStatic()).GetMasterPluginConfig()
+				Expect(err).ToNot(HaveOccurred(), "Failed to create master bridge plugin setting")
+				internalNad, err := nad.NewBuilder(APIClient, "internal", tsparams.TestNamespaceName).
+					WithMasterPlugin(masterBridgePlugin).Create()
+				Expect(err).ToNot(HaveOccurred(), "Failed to create internal NAD")
+
+				By("Configuring Local GW mode")
```

Comment:
Why is that needed?

Reply:
This is part of the multihop testing, as we need to add static routes, and workers only check the routing table if it's in local GW mode.